### PR TITLE
[User Model] Change model properties to Pascal Case

### PR DESCRIPTION
**Pull request by shepherd-l**
# Description
## One Line Summary
Renamed `Notifications`, `InAppMessages`, and `User` model properties to Pascal Case

### Motivation
Updates our code base style to follow C# coding conventions

# Testing
## Manual testing
Tested sending and opening a notification, displaying and dismissing an in app message, and opting in and out of the push subscription. App build with Android Studio 2022.1.1 and Xcode 14.2 with the OneSignal example app on a Pixel 4 simulator with Android 12 and on an iPhone 12 on iOS 15.5.

# Affected code checklist
   - [x] Notifications
      - [ ] Display
      - [ ] Open
      - [ ] Push Processing
      - [ ] Confirm Deliveries
   - [ ] Outcomes
   - [ ] Sessions
   - [x] In-App Messaging
   - [ ] REST API requests
   - [ ] Public API changes

# Checklist
## Overview
   - [x] I have filled out all **REQUIRED** sections above
   - [x] PR does one thing
   - [x] Any Public API changes are explained in the PR details and conform to existing APIs

## Testing
   - [x] I have included test coverage for these changes, or explained why they are not needed
   - [x] All automated tests pass, or I explained why that is not possible
   - [x] I have personally tested this on my device, or explained why that is not possible

## Final pass
   - [x] Code is as readable as possible.
   - [x] I have reviewed this PR myself, ensuring it meets each checklist item

- - -
This change is [Reviewable](https://reviewable.io/reviews/OneSignal/OneSignal-Unity-SDK/590)

### CODE REVIEW

File: com.onesignal.unity.core/Runtime/Notifications/Models/INotification.cs [hunk 1]
```diff
@@ -39,165 +39,165 @@ public enum LockScreenVisibility {
...
     /// <summary>
     /// If a background image was set, this object will be available
     /// </summary>
     /// <remarks>Android only</remarks>
-    [Serializable] public sealed class BackgroundImageLayout {
-        public string image;
-        public string titleTextColor;
-        public string bodyTextColor;
+    public interface IBackgroundImageLayout {
+        public string Image { get; }
```
Comment (brismithers) [inline com.onesignal.unity.core/Runtime/Notifications/Models/INotification.cs:53]: interface should not have `public`

File: com.onesignal.unity.core/Runtime/Notifications/Models/INotification.cs [hunk 2]
```diff
@@ -39,165 +39,165 @@ public enum LockScreenVisibility {
...
         /// </summary>
-        public string collapseId;
+        string CollapseId { get; }
         
         /// <summary>
         /// Gets custom additional data that was sent with the notification. Set on the dashboard under
         /// Options > Additional Data or with the data field on the REST API.
         /// </summary>
-        public Dictionary<string, object> additionalData;
+        Dictionary<string, object> AdditionalData { get; }
```
Comment (brismithers) [inline com.onesignal.unity.core/Runtime/Notifications/Models/INotification.cs:108]: you might be able to make this `IDictionary<string, object>` to provide a little more abstraction

File: com.onesignal.unity.core/Runtime/Notifications/Models/INotification.cs [hunk 3]
```diff
@@ -215,57 +215,57 @@ public enum LockScreenVisibility {
...
         /// <remarks>iOS 15+ only</remarks>
-        public string interruptionLevel;
+        string InterruptionLevel { get; }
 
         /// <summary>
         /// Attachments sent as part of the rich notification
         /// </summary>
         /// <remarks>iOS 10+ only</remarks>
-        public Dictionary<string, object> attachments;
+        Dictionary<string, object> Attachments { get; }
```
Comment (brismithers) [inline com.onesignal.unity.core/Runtime/Notifications/Models/INotification.cs:260]: you might be able to make this IDictionary<string, object> to provide a little more abstraction

File: com.onesignal.unity.core/Runtime/Notifications/Models/INotificationAction.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+using System;
+
+namespace OneSignalSDK.Notifications.Models {
+    public enum ActionType {
```
Comment (jkasten2) [inline com.onesignal.unity.core/Runtime/Notifications/Models/INotificationAction.cs:31]: ActionType has been removed in the latest spec

File: com.onesignal.unity.core/Runtime/Notifications/Models/INotificationClickedResult.cs [hunk 5]
```diff
@@ -32,15 +32,15 @@ namespace OneSignalSDK.Notifications.Models {
     /// See full documentation at
     /// https://documentation.onesignal.com/docs/sdk-notification-event-handlers#notification-opened-event
     /// </summary>
-    [Serializable] public sealed class NotificationClickedResult {
+    public interface INotificationClickedResult {
```
Comment (jkasten2) [inline com.onesignal.unity.core/Runtime/Notifications/Models/INotificationClickedResult.cs:35]: The latest spec has this as an event, the result is contained in the event instead.

File: com.onesignal.unity.core/Runtime/OneSignal.cs [hunk 6]
```diff
@@ -155,14 +155,14 @@ public abstract partial class OneSignal {
         /// </summary>
         /// <remarks>iOS Only</remarks>
         /// <returns>Awaitable boolean of whether the operation succeeded or failed</returns>
-        public abstract Task<bool> EnterLiveActivity(string activityId, string token);
+        public abstract Task<bool> EnterLiveActivityAsync(string activityId, string token);
```
Comment (jkasten2) [inline com.onesignal.unity.core/Runtime/OneSignal.cs:158]: The latest spec moved these into there own "namespace" of `LiveActivities`

**Review by jkasten2 — CHANGES_REQUESTED** [review]
Feedback is around missing the latest spec changes

**Pull request merged**